packages/bff/src/services/agreementService.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable functional/immutable-data */
+import {
+  AgreementProcessApiAgreement,
+  AgreementProcessApiResponse,
+} from "../model/api/agreementTypes.js";
+import { BffGetCatalogApiHeaders } from "../model/api/bffTypes.js";
+import { EServiceCatalogProcessApi } from "../model/api/catalogTypes.js";
+import { AgreementProcessClient } from "../providers/clientProvider.js";
+
+export const getLatestAgreement = async (
+  agreementProcessClient: AgreementProcessClient,
+  consumerId: string,
+  eservice: EServiceCatalogProcessApi,
+  headers: BffGetCatalogApiHeaders
+): Promise<AgreementProcessApiAgreement> => {
+  const getAgreementsFrom = async (
+    start: number
+  ): Promise<AgreementProcessApiResponse> =>
+    await agreementProcessClient.getAgreements({
+      headers,
+      queries: {
+        consumersIds: consumerId,
+        eservicesIds: eservice.id,
+        offset: start,
+        limit: 50,
+      },
+    });
+
+  // Fetched all agreements in a recursive way
```

Comment:
feel free to use loops if you think it may optimize resources

Reply:
I prefer to evaluate a best option possibile, considering also if is feasible to create a new specific endpoints on agreement-process API.
So I'll handle this improvement in context of [this card](https://pagopa.atlassian.net/jira/software/projects/IMN/boards/503?selectedIssue=IMN-640).